tests/unit/amo/reducers/test_collections.js

```diff
@@ -1074,4 +1075,67 @@ describe(__filename, () => {
       expect(localizedDetail).toEqual(collectionDetail);
     });
   });
+
+  describe('expandCollections', () => {
+    const username = 'some-username';
+
+    it('returns null if there is no meta passed', () => {
+      const state = reducer(undefined, {});
+
+      expect(expandCollections(state)).toEqual(null);
+    });
+
+    it('returns null if there is no collections key in meta', () => {
+      const state = reducer(undefined, {});
+      const meta = {};
+
+      expect(expandCollections(state, meta)).toEqual(null);
+    });
+
+    it('returns collection objects when ids are passed in via meta.collections', () => {
+      const firstCollection = createFakeCollectionDetail({ id: 1 });
+      const secondCollection = createFakeCollectionDetail({ id: 2 });
+      const meta = {
+        collections: [firstCollection.id, secondCollection.id],
+      };
+
+      const state = reducer(
+        undefined,
+        loadUserCollections({
+          username,
+          collections: [firstCollection, secondCollection],
+        }),
+      );
+
+      const collections = expandCollections(state, meta);
+      expect(collections[0]).toEqual(
+        createInternalCollection({ detail: firstCollection }),
+      );
+      expect(collections[1]).toEqual(
+        createInternalCollection({ detail: secondCollection }),
+      );
+    });
+
+    it('returns undefined for an unloaded collection', () => {
+      const firstCollection = createFakeCollectionDetail({ id: 1 });
+      const secondCollection = createFakeCollectionDetail({ id: 2 });
+      const meta = {
+        collections: [firstCollection.id, secondCollection.id],
+      };
+
+      const state = reducer(
+        undefined,
+        loadUserCollections({
+          username,
+          collections: [firstCollection],
+        }),
+      );
+
+      const collections = expandCollections(state, meta);
+      expect(collections[0]).toEqual(
+        createInternalCollection({ detail: firstCollection }),
+      );
+      expect(collections[1]).toEqual(undefined);
```

Comment:
Is `collections.length === 2`?

Reply:
Currently yes, but as per your comment and my response above, I'll change that.